refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
                 else:
                     missing.append(g)
             if missing:
-                _LOGGER.warning("Genomes do not include: {}".format(", ".join(missing)))
+                _LOGGER.warning(
+                    "Genomes do not include: {}".format(", ".join(missing)))
             return None if not filtered else filtered
         return genomes if not all(x in genomes for x in genome) else genome
 
+    def config_upgrade(target_version, filepath, force=False):
+        """
+            upgrade the config file to a target version, 
+            and alter genome_folder stucture 
+
+            :param str target_version: the version updated to
+            :param str filepath: file path for the current config file
+            :param bool force
+        """
```

Comment:
unindent the docstrings, I'm not sure if this will get parsed by `lucidoc` properly.

Reply:
e60d12459827a965803d17081fe766ccb84cc226